src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -52,13 +69,29 @@ export const Configurator = ({
       defaultMessage: "I don't want to use the time filter",
     }
   );
-  const [language, setLanguage] = useState<string>(() => {
-    const currentLanguage = queryString.getQuery().language;
-    if (languages.includes(currentLanguage)) {
-      return currentLanguage;
-    }
-    return languages[0];
-  });
+  const indexedViewsService = type?.indexedViewsService;
+  const [selectedIndexedView, setSelectedIndexedView] = useState<string>();
```

Comment:
technically the state could be string or undefined right?

Reply:
Fixed